model/src/main/resources/reference.yml

```diff
@@ -36,6 +36,15 @@ ort:
     patentFilenames: ['patents']
     rootLicenseFilenames: ['readme*']
 
+  # Package configurations have to be unique by ID and provenance. Ensure that different providers do not provide
+  # configurations for the same package, see https://github.com/oss-review-toolkit/ort/issues/6972 for details.
```

Comment:
I believe that the requirement "Ensure that different providers do not provide configurations for the same package" is something we cannot ask, because this cannot be done easily in an automated way. 

How about the following approach: 
1. Add a map 'provider-name -> Set<PackageConfiguration>' to `OrtResult.resolvedConfiguration`.
2. Then consumption of resolved configuration is adjusted to `merge` the path excludes and `license finding curations` of all providers, thereby removing the requirement that there is only one package configuration across all providers applicable for a package.

Reply:
> I believe that the requirement "Ensure that different providers do not provide configurations for the same package" is something we cannot ask, because this cannot be done easily in an automated way.

I agree that we cannot ask that long-term.
However, this is how ORT currently works with package-configurations. There cannot be package-configurations with the same ID and source in the .ort.yml and the global package-configurations.

Are you saying that the implementation of #6972 is a precondition for this PR?

Reply:
As this change is only about making package configuration providers plugins it should not have to address issues which already existed previously.

Reply:
I'm ok to leave this out for now.

Anyhow, am I right that ORT previously supported only a single package configuration provider, while as of this change it allows multiple? (Asking because of above argumentation: "...should not have to address issues which already existed previously")

Reply:
> am I right that ORT previously supported only a single package configuration provider, while as of this change it allows multiple?

Package-Configurations from two different sources were already applied before this PR (.ort.yml, and the one from the CLI option), if the feature was enabled:
https://github.com/oss-review-toolkit/ort/blob/dc870389cd61b4208f5252cf38c6a873d0de6e41/plugins/commands/evaluator/src/main/kotlin/EvaluatorCommand.kt#L277C6-L280